From 9c8526ae733ccb626328fcbe8f4aaa0c997b9382 Mon Sep 17 00:00:00 2001
From: Rohan Puri <rohan_puri@xyratex.com>
Date: Wed, 30 Oct 2013 18:00:16 +0530
Subject: [PATCH 116/121] Changes summary : - 1. FIXED BUG: - We can do
 m0_fop_put() with rpc items sm group lock held only.

This change was made to fix, intermittent addb-mon-infra UT failure asserting on
channels mutex lock should be held in m0_chan_fini().
---
 addb/addb_monitor.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/addb/addb_monitor.c b/addb/addb_monitor.c
index 3f8cdef..5ae0b77 100644
--- a/addb/addb_monitor.c
+++ b/addb/addb_monitor.c
@@ -446,7 +446,10 @@ static int addb_monitor_stats_fop_send(struct m0_stats_update_fop *fop_data,
 	item->ri_deadline = 0;
 
 	rc = m0_rpc_oneway_item_post(conn, item);
+	M0_ASSERT(item->ri_rmachine);
+	m0_sm_group_lock(&item->ri_rmachine->rm_sm_grp);
 	m0_fop_put(stats_update_fop);
+	m0_sm_group_unlock(&item->ri_rmachine->rm_sm_grp);
 
 	return rc;
 }
-- 
1.8.3.2

